gnd/src/main/java/com/google/android/gnd/persistence/geojson/GeoJsonParser.java

```diff
@@ -46,41 +46,32 @@
     this.uuidGenerator = uuidGenerator;
   }
 
-  /**
-   * Converts a JSONArray to an array of JSONObjects. Provided for compatibility with java8 streams.
-   * JSONArray itself only inherits from Object, and is not convertible to a stream.
-   */
-  private static JSONObject[] toArray(JSONArray arr) {
-    JSONObject[] result = new JSONObject[arr.length()];
+  private String readJsonFile(File file) {
+    try {
+      InputStream is = new FileInputStream(file);
+      BufferedReader buf = new BufferedReader(new InputStreamReader(is));
+      StringBuilder sb = new StringBuilder();
 
-    for (int i = 0; i < arr.length(); i++) {
-      try {
-        JSONObject o = arr.getJSONObject(i);
-        result[i] = o;
-      } catch (JSONException e) {
-        Log.e(TAG, "couldn't parse json", e);
+      int res = buf.read();
+      while (res != -1) {
+        sb.append(res);
+        res = buf.read();
       }
+      return sb.toString();
+    } catch (IOException e) {
+      Log.e(TAG, "Unable to load JSON", e);
     }
-
-    return result;
+    return "";
```

Comment:
Instead of eating the exception and returning "", we can throw it and catch it in `intersectingTiles`, eliminating the need for a "fake" value in the interim.

Reply:
Ah, gotcha. ✅